src/resolvelib/resolvers.py

```diff
@@ -369,11 +371,11 @@ def resolve(self, requirements, max_rounds):
                 # Backtrack if pinning fails. The backtrack process puts us in
                 # an unpinned state, so we can work on it in the next round.
                 success = self._backtrack()
+                self._failure_causes = [i for c in failure_causes for i in c.information]
```

Comment:
This line is too long, and flake8 is not happy about that.

Reply:
Also I'm suddenly unsure why I placed this _after_ backtracking. I'm going to experiment a little bit with the placement of this line and also how to make it nicely fit the 80 character limit (which I didn't realize sorry).

Reply:
It's all good. :)

FWIW, you can just change this to:

```suggestion
                self._failure_causes = [
                    i for c in failure_causes for i in c.information
                ]
```

Reply:
Sorry I accidentally deleted my previous comment. Thanks for the suggestion.

I'm going to leave it broken for just a few hours while I decide where to put the line. I realized there are 3 places it can go and they all make a difference:

1. Right after `failure_causes = ...`
2. Right after `if failure_causes`
3. Where it currently is

And they would mean

1. Limits the change in preference to only when failures are happening in the backtracking
2. The change in preference happens immediately once backtracking starts and the preference would only be changed when a new failure happened
3. Same as 2 but the preference would take effect after backtracking 

I've realized that number 3 is definitely wrong, and that it should be 1 or 2. I'm going to go back and test all my reproducible examples, post the finding to https://github.com/pypa/pip/issues/10479 with the requirements.txts attached, and then update this PR.

Reply:
This is now resolved